packages/mdc-dialog/index.ts

```diff
@@ -111,16 +62,50 @@ class MDCDialog extends MDCComponent {
     this.foundation_.setAutoStackButtons(autoStack);
   }
 
+  static attachTo(root: HTMLElement) {
+    return new MDCDialog(root);
+  }
+
+  private buttonRipples_!: MDCRipple[];
+
+  private buttons_!: HTMLElement[];
+
+  private defaultButton_!: HTMLElement | null;
+
+  private container_!: HTMLElement;
+
+  private content_: HTMLElement | null = null;
+  private initialFocusEl_: HTMLElement | null = null;
+  private focusTrapFactory_!: (
+    element: HTMLElement | string,
+    userOptions?: createFocusTrap.Options,
+  ) => createFocusTrap.FocusTrap;
+
+  private focusTrap_!: createFocusTrap.FocusTrap;
+
+  private handleInteraction_!: SpecificEventListener<'click'|'keydown'>;
+
+  private handleDocumentKeydown_!: SpecificEventListener<'keydown'>;
+  private handleOpening_!: () => void;
+
+  private handleClosing_!: () => void;
+
+  private layout_!: () => void;
+
   initialize(focusTrapFactory = createFocusTrap, initialFocusEl = null) {
-    this.container_ = /** @type {!Element} */ (this.root_.querySelector(strings.CONTAINER_SELECTOR));
+    this.container_ = this.root_.querySelector(strings.CONTAINER_SELECTOR) as HTMLElement;
+    if (!this.container_) {
+      throw new Error(`Dialog component requires a ${strings.CONTAINER_SELECTOR} container element`);
+    }
     this.content_ = this.root_.querySelector(strings.CONTENT_SELECTOR);
     this.buttons_ = [].slice.call(this.root_.querySelectorAll(strings.BUTTON_SELECTOR));
     this.defaultButton_ = this.root_.querySelector(strings.DEFAULT_BUTTON_SELECTOR);
-    this.buttonRipples_ = [];
-    this.focusTrapFactory_ = focusTrapFactory;
+    this.focusTrapFactory_ = focusTrapFactory as unknown as util.focusTrap;
```

Comment:
Remove unnecessary type casts

Reply:
Turns out we still need the `unknown` type cast. `focus-trap` exports and `.d.ts` definitions are kinda funky.